apps/synchronize/src/github/synchronizer.js

```diff
@@ -117,23 +117,24 @@ export class GitHubSynchronizer {
             organizationIdByRepositoryId[githubRepository.id] || null,
           userId: userIdByRepositoryId[githubRepository.id] || null,
           private: githubRepository.private,
+          githubDefaultBranch: githubRepository.default_branch,
         };
 
         let [repository] = await Repository.query().where({
           githubId: githubRepository.id,
         });
 
         if (repository) {
-          await repository.$query().patchAndFetch(data);
-        } else {
-          repository = await Repository.query().insert({
+          return repository.$query().patchAndFetch({
             ...data,
-            enabled: false,
-            baselineBranch: "master",
+            baselineBranch:
+              data.githubDefaultBranch !== repository.baselineBranch
+                ? repository.baselineBranch
+                : null,
           });
         }
 
-        return repository;
+        return Repository.query().insertAndFetch({ ...data, enabled: false });
```

Comment:
You replaced `insert` by `insertAndFetch`, why?

Reply:
I want to insure coherency between return cases.

Previously, this method returned not patched `repository` or `undefined`.

With my change, this method return the updated or created `repository`.

Reply:
But maybe if not need it should return undefined? Better to avoid a fetch for perf issue (but not very important).